service/chartconfig/v1/resource/chart/create.go

```diff
@@ -1,7 +1,64 @@
 package chart
 
-import "context"
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/chart-operator/service/chartconfig/v1/key"
+	"github.com/giantswarm/microerror"
+)
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	chartState, err := toChartState(createChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if chartState.ChartName != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating chart %s", chartState.ChartName))
+
+		tarballPath, err := r.apprClient.PullChartTarball(customObject)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		err = r.helmClient.InstallFromTarball(tarballPath)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created chart %s", chartState.ChartName))
+	} else {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("not creating chart %s", chartState.ChartName))
+	}
+
 	return nil
 }
+
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
+	currentChartState, err := toChartState(currentState)
+	if err != nil {
+		return ChartState{}, microerror.Mask(err)
```

Comment:
Why not dealing with a pointer and be free to return nil here?

Reply:
I'm fine either way, just followed what we had already there, WDYT @rossf7?

Reply:
I'd prefer we use a pointer so we can return nil. That's what I did for the other methods in the helm client.

Reply:
Done, thx!